core/utils.ts

```diff
@@ -148,7 +163,14 @@ export function validate(compiledGraph: dataform.ICompiledGraph): dataform.IGrap
       validationErrors.push(dataform.ValidationError.create({ message, actionName }));
       return true;
     }
-    return (action.dependencies || []).some(d => {
+    const depsFQ: string[] = [];
+    (action.dependencies || []).forEach((d: string) => {
+      const [fQd, err] = matchFQName(d, allActionNames);
```

Comment:
ditto, not sure this code is needed any longer, the dependencies are already fully-qualified and necessarily match action names.

Reply:
It was required because the test case was not well designed. I have changed the definition of the test case and now this line can go.